trtools/mergeSTR/mergeSTR.py

```diff
@@ -172,19 +179,44 @@ def GetRefAllele(current_records, mergelist):
     ref : str
        Reference allele string. Set to None if conflicting references are found.
     """
-    refs = []
+
+    def DefaultKey(record: trh.TRRecord):
```

Comment:
I believe harmonization should make it so that record.ref_allele is correct regardless of the vcftype, so you don't have to do this sort of check here. So you should just be able to pull that field. (Or if not, we should fix the harmonization to make that so).

Reply:
You are right, when I wrote this I was afraid that the REF field might contain something like <STR41>, similarly to ALT fields from some callers, but now I realize that something like that wouldn't make any sense. Thanks for pointing that out!

Reply:
I reverted the changes but found an additional problem. It looks like EH records usually have only one BP as a value in the REF column. Harmonization of EH records expands the EH record motif and fabricates this REF allele. Use of ref from the harmonized record can cause differences like this: 
```
Output line: CAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAGCAG
Control line: C
```
A possible solution is to use the old REF field from the non-harmonized VCF record and to use the old REF and ALT fields from non harmonized records (reverting most of my changes), but this might cause some weird behavior, for example on HipSTR records with and without flanking BPs.
But I guess it depends on what you think is the intended behavior. Is it OK to remove the flanking BPs during the merge or not?

Reply:
Yes, we should remove flanking bps during the merge and make the explicit in the documentation.

I had forgotten about the different representations between callers issue. You're right, it doesn't make sense use harmonized records for outputting in general, it needs to look like the original caller's output. I think the solution in the current commit is fine then - check if it's a HipSTR record, and if so use the harmonized record's alleles which have the flanking bps removed (and are sequences which is the same type of output that HipSTR produces). Otherwise just proceed with what's in the record, as this problem only pertains to HipSTR.

That all seem reasonable to you?

Reply:
Yes, it is. I've reverted my local changes and fixed the case problem from your other comment.